File: CP/armstrong.c
#include <stdio.h>
#include <math.h>

int main()
{
    int num, temp, originalNum, remainder;
    int sum = 0, n = 0;

    printf("Input a number: ");
    scanf("%d", &num);
    originalNum = num;
    if (num == 0)
    {
        n = 1;
    }
    else
    {
        for (temp = num; temp != 0; temp /= 10)
            n++;
    }

    for (temp = num; temp != 0; temp /= 10)
    {
        remainder = temp % 10;
        sum += pow(remainder, n);
    }
    if (sum == originalNum)
        printf("%d is an Armstrong number.\n", originalNum);
    else
        printf("%d is not an Armstrong number.\n", originalNum);

    return 0;
}

File: javapractise/java/armstrong.java

public class armstrong {

    public static void main(String[] args) {
        int num = 153;
        int originalNum = num;
        int remainder, result = 0;
        int digits = 0;
        while (originalNum != 0) {
            originalNum /= 10;
            digits++;
        }

        originalNum = num;

      
        while (originalNum != 0) {
            remainder = originalNum % 10;
            result += Math.pow(remainder, digits);
            originalNum /= 10;
        }

        // Step 3: Check
        if (result == num) {
            System.out.println(num + " is an Armstrong number."); 
        }else {
            System.out.println(num + " is not an Armstrong number.");
        }
    }
}
